tests/unit/source_test.py

```diff
@@ -174,3 +174,60 @@ def test_update_local_repo(self, fake_local_repo, fake_local_up_to_date, fake_re
         fake_repo.return_value = repo_mock
         Repo('github.com/app/a').update_local_repo()
         pull_mock.pull.assert_called_once_with('master')
+
+    def test_assemble_remote_path_1(self):
+        repo_url = 'file:///path/to/repo'
+        repo = Repo(repo_url)
+        expected_url = 'file:///path/to/repo'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
+
+    def test_assemble_remote_path_2(self):
+        repo_url = '/path/to/repo'
+        repo = Repo(repo_url)
+        expected_url = 'file:///path/to/repo'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
+
+    def test_assemble_remote_path_3(self):
+        repo_url = 'http://path/to/repo.git'
+        repo = Repo(repo_url)
+        expected_url = 'http://path/to/repo.git'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
+
+    def test_assemble_remote_path_4(self):
+        repo_url = 'http://path/to/repo'
+        repo = Repo(repo_url)
+        expected_url = 'http://path/to/repo.git'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
+
+    def test_assemble_remote_path_5(self):
+        repo_url = 'ssh://path/to/repo/more_stuff'
+        repo = Repo(repo_url)
+        expected_url = 'ssh://path/to/repo/more_stuff'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
+
+    def test_assemble_remote_path_6(self):
+        repo_url = 'ssh://path/to/repo:80/more_stuff'
+        repo = Repo(repo_url)
+        expected_url = 'ssh://path/to/repo:80/more_stuff'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
+
+    def test_assemble_remote_path_7(self):
+        repo_url = 'git@path/to/repo:more_stuff'
+        repo = Repo(repo_url)
+        expected_url = 'git@path/to/repo:more_stuff'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
+
+    def test_assemble_remote_path_8(self):
+        repo_url = 'path/to/repo'
+        repo = Repo(repo_url)
+        expected_url = 'ssh://git@path/to/repo'
+
+        self.assertEqual(repo.assemble_remote_path(), expected_url)
```

Comment:
Seem to be missing some combinations in tests, specifically around supplying custom users